### ISSUE TITLE
feat: Restore the built-in Call Recording feature on the Google Phone app

### ISSUE THREAD

**Issue by mina-gaid** [ReVanced/revanced-patches#3198]
### Feature description

Develop and release a patch that restores the built-in call recording feature on the official Google Phone app, where this feature is locked/hidden by Google due to regional regulations. This patch should allow users to access and use the phone recording feature regardless of their region using the Google Phone app.

The functionality already exists on the Google Phone app but is now locked/hidden in certain regions or devices, as per the official documentation: 
https://support.google.com/phoneapp/answer/9803950?hl=en

Certain markets, such as India, still have access to the Call Recording feature.

The Pixel line up offered Call Recording via the Google Phone app until it was removed (hidden/locked) following an update.

Attached - Screenshot of call recording on the Google Phone app before it was removed:

![google-phone-call-recording-xda-1](https://github.com/ReVanced/revanced-patches/assets/15611727/8a9354aa-1ef8-441e-8e49-b4f0978b5acf)

### Motivation

Many users require access to their device’s built-in call recording feature for legitimate purposes. However, due to regional regulations, some OEMs lock this feature, limiting the user’s ability to fully utilize their device. By enabling this feature, we can improve the user experience by adding comprehensive control over our devices.

On some devices, such as OnePlus, the call recoding feature can be unlocked via an ADB Command (Non-Root). However, this doesn't work on all phones, such as the Pixel range because every OEM has their own modified version of the stock Android phone app. By specifically patching the official Google phone app, all Android users can be catered for regardless of device because the app is available to download on the Play store - and more often than not, it's most people's default choice.

### Acknowledgements

- [X] This request is not a duplicate of an existing issue.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

**Comment by KobeW50** [comment]
If mounting the patched app will be required for this, I suggest creating a similar patch for other popular calling apps which wouldn't need mounting (in order to help out the non-root users).

**Comment by mina-gaid** [comment]
It wouldn't be practical due to maintenance concerns. I previously made a similar request that applied to all the default stock phone apps by major OEMs. However, it was pointed out that these apps aren't as similar as one might think. Admittedly, I don't have much knowledge about Android apps.

The reason it wouldn't work is that not all devices have the necessary built-in capabilities. To the best of my knowledge, only OnePlus, Samsung, and certain Pixel phones offer this feature, and it's either region-locked or completely removed after an update.

The most viable solution is to patch the Google phone app, which is the default for Pixel phones and is also available on the Play Store for other devices. This feature was also once available in the Google phone app for Pixel phones before it was removed.

I want to emphasize that I'm aware of third-party apps that offer this capability, but they are impractical, riddled with intrusive ads, and often require a monthly subscription. Restoring this feature to the Google phone app would be a fantastic solution and would cater to all Android phones regardless of make or model.

**Comment by dib12** [comment]
@mina-gaid call recording can be enable easily if u have this XML file.
If rooted copy to product/etc/permission. And check.

https://dumps.tadiphone.dev/dumps/redmi/sky/-/blob/missi_phone_global-user-13-TKQ1.221114.001-V14.0.5.0.TMWINXM-release-keys/product/etc/permissions/com.google.android.apps.dialer.call_recording_audio.xml

Revanced patch idea also good.
Let's wait for dev to add this feature

**Comment by mina-gaid** [comment]
@dib12 Yeah, Googledialer call recording announcement is annoying, but as you said, that can be a separate patch.

I personally don't mind the announcement, as I only need it when making calls with Amazon support, bank etc who are also recording me. Most support agents use to think it was hilarious that I was recording them back. 😂
